bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/DialogProcessor.java

```diff
@@ -282,15 +283,24 @@ public synchronized void sttEventReceived(STTEvent sttEvent) {
                 SpeechRecognitionEvent sre = (SpeechRecognitionEvent) sttEvent;
                 String question = sre.getTranscript();
                 logger.debug("Text recognized: {}", question);
-                try {
-                    toggleProcessing(false);
-                    String answer = hli.interpret(locale, question);
-                    logger.debug("Interpretation result: {}", answer);
-                    say(answer);
-                } catch (InterpretationException e) {
-                    logger.debug("Interpretation exception: {}", e.getMessage());
-                    say(e.getMessage());
+                toggleProcessing(false);
+                String answer = null;
+                String error = null;
+                for (var interpreter : hlis) {
+                    try {
+                        answer = interpreter.interpret(locale, question);
+                        logger.debug("Interpretation result: {}", answer);
+                        error = null;
+                        break;
+                    } catch (InterpretationException e) {
+                        logger.debug("Interpretation exception: {}", e.getMessage());
+                        var msg = e.getMessage();
+                        if (msg != null) {
+                            error = msg;
```

Comment:
If there is more than one error message, earlier messages are discarded. Is that intentional?

Reply:
Yes, this is done in order to find the first interpreter which returns a successful response. 

Maybe this can be improved in a separate pr by adding a subclass of the general Interpretation Exception to ensure those are unsuccessful responses instead of any other execution error. WDYT?